src/main/java/org/spongepowered/api/service/pagination/PaginationList.java

```diff
@@ -0,0 +1,186 @@
+     *
+     * @param source The source to send to
+     */
+    void sendTo(MessageReceiver source);
+
+    /**
+     * Send the constructed pagination list to the specified {@link MessageChannel}.
+     * @param channel
+     */
+    default void sendTo(MessageChannel channel) {
+        for (MessageReceiver receiver : channel.getMembers()) {
+            this.sendTo(receiver);
+        }
+    }
+
+    /**
+     * Builds a paginated output for an iterable of {@link Text}s.
+     */
+    interface Builder extends ResettableBuilder<PaginationList, Builder> {
+
+        /**
+         * Set the contents of this output as an Iterable.
+         * If this {@link Iterable} is a {@link List}, bidirectional navigation is supported.
+         * Otherwise, only going to the next page will be supported
+         *
+         * @param contents The contents to output
+         * @return this
+         */
+        Builder contents(Iterable<Text> contents);
+
+        /**
+         * Set the contents of this output to be the given array of contents.
+         *
+         * @param contents The contents to output
+         * @return this
+         */
+        Builder contents(Text... contents);
+
+        /**
+         * Set the title text to be used in the title bar of this pagination.
+         * This should be less than one line long.
+         *
+         * @param title The title to use.
+         * @return this
+         */
+        Builder title(Text title);
+
+        /**
+         * Set the header to be displayed for this output on all pages after the title bar but before the contents
+         * Header and footer will use this Text's style and color for formatting.
+         *
+         * @param header The header to set
+         * @return this
+         */
+        Builder header(Text header);
+
+        /**
+         * Set the footer to be displayed for this output on all pages after the contents and page navigation bar.
+         *
+         * @param footer The footer to set
+         * @return this
+         */
+        Builder footer(Text footer);
+
+        /**
+         * Set the padding character to be used when centering headers and footers.
+         *
+         * @param padding The padding to use
+         * @return this
+         */
+        Builder padding(Text padding);
+
+        /**
+         * Creates a {@link PaginationList} from this pagination builder.
+         *
+         * @return The pagination list
+         */
+        PaginationList build();
+
+        /**
+         * Send the constructed pagination list to the given source.
+         *
+         * @param source The source to send to
+         */
+        void sendTo(MessageReceiver source);
+
+        /**
+         * Send the constructed pagination list to the specified {@link MessageChannel}.
+         * @param channel
+         */
+        default void sendTo(MessageChannel channel) {
+            for (MessageReceiver receiver : channel.getMembers()) {
```

Comment:
The thing is, the pagination list needs access to each `MessageReciever`, so it can get the `PaginationCalculator` for it. Note that this is a default method, so a plugin-provided `PaginationService` could just override this.